metaflow/plugins/parallel_decorator.py

```diff
@@ -122,8 +122,24 @@ def _local_multinode_control_task_step_func(flow, env_to_use, step_func, retry_c
     os.environ["MF_PARALLEL_NODE_INDEX"] = "0"
     step_func()
 
-    # join the subprocesses
-    for p in subprocesses:
-        p.wait()
+    # join the subprocesses and store logs
+    flow_datastore = flow._datastore.parent_datastore
+    errors = []
+    for task_id, p in subprocesses:
+        stdout, stderr = p.communicate()
+
+        print("--- Task {} stdout: ---".format(task_id))
+        print(stdout.decode("utf8"))
+        print("--- Task {} stderr: ---".format(task_id))
+        print(stderr.decode("utf8"))
+
+        task_datastore = flow_datastore.get_task_datastore(
```

Comment:
Suggest using the `task_pre_step` hook to get the `task_datastore` (it's one of the parameters) and using it directly. This would also set the proper attempt ID for you (which may be incorrect here since you always use 0). To be consistent with runtime.py:829, I think we should also use RUNTIME_LOG_SOURCE.

Reply:
I fixed the attempt.  Using the task_pre_step won't work in this case, because it is the launcher task that needs to access the logs of the launched subtasks.